jetbrains-core/src/software/aws/toolkits/jetbrains/services/clouddebug/actions/InstrumentDialogWrapper.kt

```diff
@@ -56,9 +59,23 @@ class ConfirmNonProductionDialogWrapper(project: Project, serviceName: String) :
     override fun createSouthAdditionalPanel() = JPanel(GridBagLayout()).apply { add(doNotShowAgain) }
 
     override fun doOKAction() {
+        recordTelemetry(TelemetryConstants.TelemetryResult.Succeeded)
         if (doNotShowAgain.isSelected) {
             settings.showEnableDebugWarning = false
         }
         super.doOKAction()
     }
+
+    override fun doCancelAction() {
+        recordTelemetry(TelemetryConstants.TelemetryResult.Cancelled)
+        super.doCancelAction()
+    }
+
+    private fun recordTelemetry(result: TelemetryConstants.TelemetryResult) {
+        telemetry.record(project) {
+            datum("${TelemetryConstants.CLOUDDEBUG_TELEMETRY_PREFIX}_confirm_enable") {
```

Comment:
Our current telemetry scheme is prefix_metricname, with no underscores or other punctuation in the metric name itself.

We're also not camelcasing or anything, but that's a standardization rule we were saying we might ditch. Optional if you want to camel case it, I won't knock you if you prefer that.